src/sqlfluff/core/dialects/dialect_ansi.py

```diff
@@ -957,14 +957,19 @@ class CaseExpressionSegment(BaseSegment):
             OneOf(
                 Sequence(
                     OneOf(
-                        Ref("BinaryOperatorGramar"),
+                        Ref("BinaryOperatorGrammar"),
                         Sequence(
                             Ref.keyword("NOT", optional=True),
                             OneOf("LIKE", "RLIKE", "ILIKE"),
                         )
                         # We need to add a lot more here...
                     ),
                     Ref("Expression_A_Grammar"),
+                    Sequence(
+                        Ref.keyword("ESCAPE"),
+                        Ref("Expression_A_Grammar"),
```

Comment:
@alanmcruickshank this fixes the issue but I'd love to add some comments to describe what Expression_A_Grammar, Expression_B_Grammar etc. are exactly. Could you explain? 🤔 Or is this due for an overhaul at some point anyway?

Reply:
Great question. Probably due to an overhaul. The naming goes waaaay back to when I wholesale copied chunks of the sql grammar from the excellent docs of CockroachDB: https://www.cockroachlabs.com/docs/v20.2/sql-grammar.html#a_expr

The naming is copied almost entirely from there.

Now that the grammar has stabilised a little, I would definitely be up for better names for these elements. Feel free to raise a PR to rename them if you've got better ideas.